ros/api/v1/suggested_instance_types.py

```diff
@@ -96,3 +109,104 @@ def build_sort_expression(self, order_how, order_method):
 
         abort(403, message="Unexpected sort method {}".format(order_method))
         return None
+
+
+class SuggestedInstanceDetails(Resource):
+    performance_utilization_fields = {
+        'cpu': fields.Integer,
+        'memory': fields.Integer,
+        'max_io': fields.Float,
+        'io_all': fields.Raw
+    }
+
+    data = {
+        'fqdn': fields.String,
+        'display_name': fields.String,
+        'instance_type': fields.String,
+        'os': fields.String,
+        'performance_utilization': fields.Nested(performance_utilization_fields),
+        'state': fields.String,
+        'last_reported': fields.DateTime(dt_format='iso8601')
+    }
+
+    instance_type_detail = {
+        'instance_type': fields.String,
+        'cloud_provider': fields.String,
+        'description': fields.String
+    }
+
+    meta_fields = {
+        'count': fields.Integer,
+        'limit': fields.Integer,
+        'offset': fields.Integer,
+        'suggested_instance_type_detail': fields.Nested(instance_type_detail)
+    }
+
+    links_fields = {
+        'first': fields.String,
+        'last': fields.String,
+        'next': fields.String,
+        'previous': fields.String
+    }
+    output = {
+        'meta': fields.Nested(meta_fields),
+        'links': fields.Nested(links_fields),
+        'data': fields.List(fields.Nested(data))
+    }
+
+    @marshal_with(output)
+    def get(self, instance_type):
+        limit = limit_value()
+        offset = offset_value()
+        org_id = org_id_from_identity_header(request)
+        system_query = group_filtered_query(system_ids_by_org_id(org_id))
+        query = db.session.query(
+            System.fqdn,
+            System.display_name,
+            func.concat(System.operating_system["name"].astext, ' ', System.operating_system['major'], '.',
```

Comment:
We have [this](https://github.com/RedHatInsights/ros-backend/blob/main/ros/lib/models.py#L90) function already to deserialize OS data. We can use that.

Reply:
While trying to use the `deserialize_host_os_data` function I faced below challenges,
1. Its the property of System's model
2. If I want to use it then in query I would need to select all columns of System's model(actually call whole model) even though I don't need all of them

So I have below thoughts,
a. We retrieve all columns from tables and use the function, which I think we should not do?
b. Rewrite the `deserialize_host_os_data` but not as property, this means we would also change existing code which depends on this method.

Let me know your thoughts

Reply:
The second one is better. But I don't know much changes are required in the existing code. We can target it in separate PR or in the same. What do you think?

Reply:
As per the discussion we would like to take route b, which would need more code change. Considering this I would have separate pr for that refactoring